bigbluebutton/bigbluebutton
# Remove unused FreeSWITCH modules from configuration

### What does this PR do?

Remove FreeSWITCH configuration for modules which are of no use in the context of BigBlueButton. This patch removes most files from `autoload_config` and reduces the `modules.conf.xml` to the following lines:

```xml
<configuration name="modules.conf" description="Modules">
  <modules>
    
    <load module="mod_console"/>
    <load module="mod_logfile"/>

    
    <load module="mod_event_socket"/>

    
    <load module="mod_sofia"/>

    
    <load module="mod_commands"/>
    <load module="mod_conference"/>
    <load module="mod_dptools"/>

    
    <load module="mod_dialplan_xml"/>

    
    <load module="mod_opus"/>
    <load module="mod_opusfile"/>

    
    <load module="mod_sndfile"/>
    <load module="mod_native_file"/>

  </modules>
</configuration>
```

The following `tree` remains:

```
$ tree 
.
├── autoload_configs
│   ├── acl.conf.xml
│   ├── conference.conf.xml
│   ├── console.conf.xml
│   ├── event_socket.conf.xml
│   ├── logfile.conf.xml
│   ├── modules.conf.xml
│   ├── opus.conf.xml
│   ├── sndfile.conf.xml
│   ├── sofia.conf.xml
│   ├── switch.conf.xml
│   └── timezones.conf.xml
├── dialplan
│   ├── default
│   │   ├── bbb_conference.xml
│   │   ├── bbb_echo_test.xml
│   │   └── bbb_echo_to_conference.xml
│   ├── default.xml
│   ├── public
│   │   ├── bbb_sip.xml
│   │   └── bbb_webrtc.xml
│   └── public.xml
├── directory
│   ├── default
│   │   └── bbbuser.xml
│   └── default.xml
├── freeswitch.xml
├── sip_profiles
│   ├── external
│   │   └── example.xml
│   ├── external-ipv6
│   │   └── example.xml
│   ├── external-ipv6.xml
│   ├── external.xml
│   ├── internal-ipv6.xml
│   └── internal.xml
└── vars.xml

9 directories, 28 files
```

### Motivation

A slim FreeSWITCH configuration reduces the attack surface.

### Manual test instructions

1. Perform a fresh install of BBB on some server
2. Ensure that everything works (including Audio, Video, Screensharing, Recording, etc.)
3. Apply this patch (as root):
```
curl -LO https://patch-diff.githubusercontent.com/raw/bigbluebutton/bigbluebutton/pull/11864.patch
patch -d /opt/freeswitch -p4 < 11864.patch
```
4. Restart bbb (as root) `bbb-conf --restart`
5. Check that everything still works (including Audio, Video, Screensharing, Recording, etc.)
